db/db_impl/db_impl.cc

```diff
@@ -1083,10 +1092,20 @@ Status DBImpl::SetDBOptions(
           mutex_.Lock();
         }
         if (new_options.stats_dump_period_sec > 0) {
+          // In case many DBs have `stats_dump_period_sec` enabled in rapid
+          // succession, we can have all threads dumping at once, which causes
+          // severe lock contention in jemalloc. Ensure successive enabling of
+          // `stats_dump_period_sec` are staggered by at least one second in the
+          // common case.
+          static unsigned int stats_dump_threads_started = 0;
```

Comment:
I have two questions:
1.  Is it really intended to have a different "stats_dump_threads_started" count for SetDBOptions than from StartTimedTasks?  Should the value be shared and static to the file, rather than to each method.

2.  In the case of SetDBOptions, should this "stated" delay always be there, or only there if the task was not scheduled previously?

Reply:
1. I suppose they can be independent, as I'm not worried about the case where enabling a stats dump thread via DB open and enabling it by `SetOptions()` are called in rapid succession.
2. I think always is fine. The users of `SetDBOptions()` often apply a dynamic option change to all their DBs in rapid succession, so it seems important to stagger every time we start a stats dump thread.